src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,101 @@
+const DIGIT_BLOCK_SIZE        = 36;
+const DIGIT_BLOCK_LEFT_OFFSET = 66;
+const DIGIT_BLOCK_CORRECTION  = 2;
+
+const DigitTicker = (() => {
+    let $container, source;
+    const digits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+    const init   = (container_id, contract) => {
+        source = contract;
+
+        $container = $(`#${container_id}`);
+        $container
+            .append($('<div />', { class: 'peek-box' })
+                .append($('<div />', { class: 'mask', text: '0/0' }))
+                .append($('<div />', { class: 'peek' })))
+            .append($('<div />', {
+                class: 'digits', html : digits.map(digit => `<div class='digit digit-${digit}'>${digit}</div>`).join(''),
+            }));
+        colorWinningBoxes(winningNumbers(contract));
```

Comment:
if we already have source as global, why do we need to pass contract around?

Reply:
The need for global contract is resolved.